Adding new and missing schema files for Automation

vrdmr opened the pull request:
Contains the following changes:
- Added the missing schema files for resources in the `2015-10-31` stable resources.
- Added the corresponding examples and fixed the existing ones.

vrdmr commented:
FYI @finiteattractor

vrdmr commented:
@Tiano2017 Could you please review this PR and merge it? Thanks.

Tiano2017 commented:
@vrdmr Will take a look.

TianoMS commented on schemas/2014-04-01-preview/deploymentTemplate.json:
```diff
@@ -211,14 +211,21 @@
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/variables" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/schedules" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobs" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobSchedules" },
```
is this going to break existing templates?

vrdmr replied:
The earlier schema was hand-written and old (2 years ago). I generated the new ones from Autorest and put renamed them here. The resources themselves haven't changed, but I can double check with my team to make sure that this deploymentTemplate does not break existing templates.

vrdmr replied:
Hi @TianoMS,

I tested with existing templates as well as a custom template I made and based on my testing, the changes are backward compatible.

```
varads-mbp:tools varadmeru$ npm test

> template-schema-validation-tools@1.0.0 test /Users/varadmeru/work/microsoft/github/azure-resource-manager-schemas/tools
> mocha runSchemaValidation.js

  validate template files with schema - 
    ✓ running schema validation on 'Template2016.json' (665ms)
    ✓ running schema validation on 'Template2016.preview.json' (265ms)
    ✓ running schema validation on 'TestTemplate.json' (1020ms)
    ✓ running schema validation on 'TestTemplate.preview.json' (807ms)
    ✓ running schema validation on 'Various.json' (289ms)

  5 passing (3s)

```

TianoMS commented on tests/2017-05-15-preview/Microsoft.Automation.tests.json:
```diff
@@ -0,0 +1,4 @@
+{
+    "tests": [
+    ]
```
add tests here?

vrdmr replied:
The `2017-05-15-preview` version is currently being added and there are no tests for the same right now. I'll work with the team to populate the tests. I'll remove this file and once the tests are in, put the file back in.

vrdmr replied:
Added the tests here.

Tiano2017 commented:
@vrdmr left a couple of comments. Can you take a look and address them?

vrdmr commented:
@Tiano2017 I've fixed the tests and added a test in `tests/2017-05-15-preview/Microsoft.Automation.tests.json`. Could you please take a look.

Thanks,
Varad
